UPGRADE.rst

```diff
@@ -85,6 +85,27 @@ for example:
      wget https://packages.matrix.org/debian/pool/main/m/matrix-synapse-py3/matrix-synapse-py3_1.3.0+stretch1_amd64.deb
      dpkg -i matrix-synapse-py3_1.3.0+stretch1_amd64.deb
 
+Upgrading to v1.35.0
+====================
+
+Deprecation of the built-in account validity feature
+----------------------------------------------------
+
+As of Synapse v1.35.0, the built-in account validity feature is deprecated in favour of
+plugin modules. If you are using this feature and want to keep its current behaviour,
+please use the `synapse-email-account-validity <https://github.com/matrix-org/synapse-email-account-validity>`_
+module.
+
+Synapse's built-in account validity feature will be removed in a future release.
+
+
+Removal of account validity template configuration settings
+-----------------------------------------------------------
+
+In line with previous changes around templates management in Synapse, support has been
+removed for the configuration settings ``template_dir``, ``account_renewed_html_path`` and
+``invalid_token_html_path`` in the ``account_validity`` section of the configuration file.
+
```

Comment:
... which previous changes? I don't remember us doing this sort of thing before.

Reply:
I seemed to remember we had done work to hardcode the name of templates in the past, but looking at other places in the code it looks like I've imagined that.

Reply:
as a general rule we have hardcoded names for new templates. Where we have existing config settings, we have removed them from the sample config, but they continue to work. I'd suggest doing the latter here.

likewise `template_dir`: I would remove it from the sample config, and have it default to the global `template_dir`, but keep it working if someone has deliberately set it.